cms/static/js/views/course_info_update.js

```diff
@@ -17,32 +23,53 @@ define(['js/views/validation', 'codemirror', 'js/models/course_update',
 
             initialize: function() {
                 this.template = this.loadTemplate('course_info_update');
-                this.render();
-            // when the client refetches the updates as a whole, re-render them
+                // Disable the new update button while we load the collection.
+                this.$el.find('.new-update-button').prop('disabled', true);
+
+                // when the client refetches the updates as a whole, re-render them
                 this.listenTo(this.collection, 'reset', this.render);
                 this.listenTo(this.collection, 'invalid', this.handleValidationError);
             },
 
             render: function() {
-            // iterate over updates and create views for each using the template
-                var updateEle = this.$el.find('#course-update-list');
-            // remove and then add all children
-                $(updateEle).empty();
-                var self = this;
-                this.collection.each(function(update, index) {
-                    try {
-                        CourseInfoHelper.changeContentToPreview(
-                        update, 'content', self.options['base_asset_url']);
-                    // push notification is always disabled for existing updates
-                        var newEle = self.template({updateModel: update, push_notification_enabled: false});
-                        $(updateEle).append(newEle);
-                        DateUtils.setupDatePicker('date', self, index);
-                        update.isValid();
-                    } catch (e) {
-                    // ignore
+                // iterate over updates and create views for each using the template
+                var updateList = this.$el.find('#course-update-list'),
+                    self = this;
+
+                $(updateList).empty();
+                if (this.collection.length > 0) {
+                    this.collection.each(function(update, index) {
+                        try {
+                            CourseInfoHelper.changeContentToPreview(
+                                update, 'content', self.options.base_asset_url);
+                            // push notification is always disabled for existing updates
+                            HtmlUtils.append(
+                                updateList,
+                                HtmlUtils.HTML(self.template({updateModel: update, push_notification_enabled: false}))
+                            );
+                            DateUtils.setupDatePicker('date', self, index);
+                            update.isValid();
+
+                            if (index === self.collection.length - 1) {
+                                // Once the collection is loaded show the button.
+                                self.$el.find('.new-update-button').prop('disabled', false);
```

Comment:
This needs to happen when the list is completely loaded and this function seems to be running asynchronous to the rest of the code in render.  Therefore when we get to line 63 we have not finished the processing here in this method.   

That being said I could move it to earlier in the same function, but if I pull it out I believe I will have the same issue as before, New update button enabled before the collection was loaded.  Lastly, if I move the code on 63 outside the else it will enable the button before the collection is loaded.